fastlane/lib/fastlane/actions/cocoapods.rb

```diff
@@ -14,7 +14,7 @@ def self.run(params)
           cmd << ["cd '#{podfile_folder}' &&"]
         end
 
-        cmd << ['bundle exec'] if File.exist?('Gemfile') && params[:use_bundle_exec]
+        cmd << ['bundle exec'] if params[:use_bundle_exec] && shell_out_should_use_bundle_exec?
```

Comment:
Still not 100% satisfied with this. I'd go for 

```ruby
def self.shell_bundle_exec_prefix
    shell_out_should_use_bundle_exec ? "bundle exec" : ""
end
```

and then do 

```ruby
cmd << shell_bundle_exec_prefix if params[:use_bundle_exec]
```

everywhere (or something).